mlrun/model_monitoring/db/tsdb/timescaledb/utils/timescaledb_dataframe_processor.py

```diff
@@ -0,0 +1,140 @@
+# Copyright 2025 Iguazio
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import Optional
+
+import pandas as pd
+
+import mlrun.common.schemas.model_monitoring as mm_schemas
+from mlrun.model_monitoring.db.tsdb.timescaledb.timescaledb_connection import (
+    QueryResult,
+)
+
+
+class TimescaleDBDataFrameProcessor:
+    """Utility class for common DataFrame processing operations."""
+
+    @staticmethod
+    def from_query_result(result: Optional[QueryResult]) -> pd.DataFrame:
+        """
+        Create a DataFrame from a QueryResult object.
+
+        :param result: QueryResult object from TimescaleDB connection
+        :return: pandas DataFrame
+        """
+        return pd.DataFrame(
+            result.data if result else [], columns=result.fields if result else []
+        )
+
+    @staticmethod
+    def apply_column_mapping(
+        df: pd.DataFrame, mapping_config: dict[str, str]
+    ) -> pd.DataFrame:
+        """
+        Apply column name mapping to a DataFrame.
+
+        :param df: Input DataFrame
+        :param mapping_config: Dictionary mapping old column names to new names
+        :return: DataFrame with renamed columns
+        """
+        if df.empty or not mapping_config:
+            return df
+
+        if valid_mapping := {
+            old: new for old, new in mapping_config.items() if old in df.columns
+        }:
+            df = df.rename(columns=valid_mapping, inplace=False)
```

Comment:
No need to mention `inplace=False`, or we'd be mentioning it just about everywhere.
```suggestion
            df = df.rename(columns=valid_mapping)
```

Reply:
will fix